src/libraries/Pool.sol

```diff
@@ -515,12 +515,13 @@ library Pool {
 
         uint128 liquidityGrossBefore;
         int128 liquidityNetBefore;
+
         assembly ("memory-safe") {
             // load first slot of info which contains liquidityGross and liquidityNet packed
             // where the top 128 bits are liquidityNet and the bottom 128 bits are liquidityGross
             let liquidity := sload(info.slot)
             // slice off top 128 bits of liquidity (liquidityNet) to get just liquidityGross
-            liquidityGrossBefore := shr(128, shl(128, liquidity))
+            liquidityGrossBefore := and(liquidity, 0xffffffffffffffffffffffffffffffff)
```

Comment:
I think the reason it doesn't save gas is the parameter stack ordering.

I suspect if you did
```
liquidityGrossBefore := and(0xffffffffffffffffffffffffffffffff, liquidity)
```
then it would save gas

Reply:
another explanation for it being the same could be that solc's optimization was already doing this, similar to #750